### DISCUSSION TITLE
Velero + Restic Backups: location of kubelet pods

### DISCUSSION THREAD

**Discussion by JustinGuese** [kube-hetzner/terraform-hcloud-kube-hetzner#346] [Q&A] [answered]
Hey everyone, I'm trying to figure out the setup of velero + restic...
Restic does currently not work, but I guess I am just missing the correct /kubelet/pods path?

I have installed velero with

```
velero install \
  --use-restic \
  --provider aws \
  --plugins velero/velero-plugin-for-aws \
  --bucket <bucket> \
  --secret-file ./s3creds \
  --backup-location-config region=eu-central-1,s3ForcePathStyle="true",s3Url=https://s3somewhere,region=eu-central-1 \
  --snapshot-location-config region=eu-central-1
```

What helped with restic backups in a Microk8s environment was to patch the kubelet directory in the restic pod 
`kubectl -n velero patch daemonset.apps/restic --type='json' -p='[{"op": "replace", "path": "/spec/template/spec/volumes/0/hostPath/path", "value":"/var/snap/microk8s/common/var/lib/kubelet/pods"}]'`

I guess the same is required for THKH to work?

Or am I missing something?

The basic velero backup works, but not the PVC backup / restic part.

Is it `/var/lib/kubelet/`? But I think this should be the default anyways, so the question then would be what the error is that prevents restic to work 

**Trade offer**: If you help me getting it to work I can write a doc in the Readme about backup & restore ;)

**Comment by mysticaltech** [comment, marked as answer]
@JustinGuese Please, instead of installing with the CLI, just use Helm as in this example; it will most probably work out of the box! https://picluster.ricsanfre.com/docs/backup/#velero-installation-and-configuration

**Reply by JustinGuese** [reply]
wow quick reply ;)! let me try and I'll update you!

**Reply by JustinGuese** [reply]
i can actually confirm that using the helm chart instead solved the issue... weird! :D thx @mysticaltech

**Reply by mysticaltech** [reply]
Good to hear!! 🙏

The way I saw it is that the helm chart runs from within, so it can deduce all the needed parameters; as for the CLI, it's executed from outside, so it does not have access to such "privy" info. So you have to configure it correctly, which is harder and more time-consuming!
